### PR TITLE
RHEL8: Require depmod path based on local path

### PR DESCRIPTION
**Pull request by ofaaland**
### Motivation and Context

RHEL8 moved depmod from /sbin/depmod to /usr/sbin/depmod.  The specfile
for builds targeting RHEL8 must use the correct path for the Requires
line.

See #8724 

### Description

Assume that if the builder is targeting a RHEL8 derived system, they are
building on one.  Use the local path to depmod to indicate which path
should be used.

### How Has This Been Tested?

local builds on rhel 7 and rhel 8

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: scripts/kmodtool [hunk 1]
```diff
@@ -144,6 +144,12 @@ print_rpmtemplate_per_kmodpkg ()
 	local kernel_uname_r=${1}
 	local kernel_variant="${2:+-${2}}"
 
+	# With RHEL 8, depmod moved from /sbin to /usr/sbin
```
Comment (mdiep25) [inline scripts/kmodtool]: I find that in RHEL7.7 and RHEL8, they both symlink to ../bin/kmod
lrwxrwxrwx. 1 root root 11 Feb 10  2019 /sbin/depmod -> ../bin/kmod
lrwxrwxrwx. 1 root root 11 Feb 10  2019 /usr/sbin/depmod -> ../bin/kmod

Comment (behlendorf) [reply scripts/kmodtool]: That's right.  Either path can be used to execute `depmod` during the install.  The issue being addressed here is in the RPM packaging where the real path, and not a symlink, must be specify to set the correct dependency.

File: scripts/kmodtool [hunk 2]
```diff
@@ -144,6 +144,12 @@ print_rpmtemplate_per_kmodpkg ()
...
+	local depmod_path=/sbin/depmod
+	if [ ! -f ${depmod_path} ]; then
+		depmod_path=/usr/sbin/depmod
```
Comment (behlendorf) [inline scripts/kmodtool]: It looks as is we should also be using `depmod_path` in the `%post` and `%postun` sections.

Comment (mdiep25) [inline scripts/kmodtool]: given that /sbin/depmod and /usr/sbin/depmod both symlinked to /usr/bin/kmod, is it true that there will always be /sbin/depmod file and depmod_path will never be /usr/sbin/depmod?

Comment (behlendorf) [reply scripts/kmodtool]: Yes and no.  This is true for CentOS/RHEL 7 and 8, plus any semi-recent Fedora release.  It's not the case for CentOS/RHEL 6, which is ancient, but I'd really prefer not to break if at all possible.  I'm also unsure of the situation on other platforms so my feeling is a minimal change is desirable.

@mdiep25 by chance were you able to test this patch and verify it does resolve the packaging issue?

Comment (mdiep25) [reply scripts/kmodtool]: yes, I have and it didn't fix the problem. rpm -ihv works but not yum/dnf. I verified it by make the Requires = kmod and it worked. see https://review.whamcloud.com/#/c/36095/5/contrib/lbuild/rhel8/zfs_issue_8724.patch

Comment (behlendorf) [reply scripts/kmodtool]: That's a pretty nice way to handle it.  I think CentOS/RHEL 6 is the distro still providing depmod as part of the `module-init-tools` package, so what do you think of something like this in addition to what you've proposed.

```
%if 0%{?rhel} == 6 || 0%{?centos} == 6
Requires(post):   module-init-tools
Requires(postun): module-init-tools
%else
Requires(post):   kmod
Requires(postun): kmod
%endif
```

Comment (mdiep25) [reply scripts/kmodtool]: Thanks, I was just trying to test it. yeah, your proposed change looks good to me

**Comment by behlendorf** [comment]
@mdiep25 I've opened #9310 as a proposed alternative fix for the CI to test.  It does still need some manual testing and review for RHEL8.

**Comment by behlendorf** [comment]
Closing in favor of #9310

**Pull request closed without merging**